osf/models/institution.py

```diff
@@ -17,19 +17,44 @@ class Institution(Loggable, base.ObjectIDMixin, base.BaseModel):
     }
     # /TODO DELETE ME POST MIGRATION
 
-    # TODO Remove null=True for things that shouldn't be nullable
-    banner_name = models.CharField(max_length=255, null=True, blank=True)
+    # TODO Remove null=True for things that shouldn't be nullable POST MIGRATION
+    # e.g. CharFields should never be null=True
+
+    name = models.CharField(max_length=255)
+    description = models.TextField(blank=True, null=True, default='')
+
+    # TODO Could `banner_name` and `logo_name` be a FilePathField?
+    # Both banner_name and logo_name are the name of the file under the `banners/` or `shields/` folder in
+    # `website/static/img/institutions/`. The file must exists for OSF dashboard page to load.
+    banner_name = models.CharField(max_length=255, blank=True, null=True)
+    logo_name = models.CharField(max_length=255, blank=True, null=True)
+
+    # The protocol used to delegate authentication: `CAS`, `SAML`, `OAuth`, e.t.c
+    # For our institution, we use shibbloeth's implementation for SAML and pac4j's implementation for CAS and OAuth
+    # Only institutions with a delegation protocol shows up on institution login page
+    DELEGATION_PROTOCOL_CHOICES = (
+        ('cas-pac4j', 'CAS by pac4j'),
+        ('saml-pac4j', 'SAML by pac4j'),
+        ('oauth-pac4j', 'OAuth by pac4j'),
+        ('saml-shib', 'SAML by shibboleth'),
+        ('', 'No Delegation Protocol'),
+    )
+    delegation_protocol = models.CharField(max_length=15, choices=DELEGATION_PROTOCOL_CHOICES, blank=True, default='')
+
```

Comment:
Use `enum` and `choices` for delegation protocol. The "`<protocol>-<implementation>`" format makes it easier for us to understand. Use empty string `''` for non-delegation institutions.